lib/matplotlib/pyplot.py

```diff
@@ -1185,6 +1185,11 @@ def close(fig: None | int | str | Figure | Literal["all"] = None) -> None:
         - ``str``: a figure name
         - 'all': all figures
 
+    Notes
+    -----
+    pyplot maintains a reference to figures created with `figure()` until
+    `close()` is called on them.  Closing figures is thus necessary to prevent
+    running out of memory (see also :rc:`figure.max_open_warning`).
```

Comment:
Can we get this more in the direction that `close()` is the explicit way to deregister a figure from pyplot? I want to prevent the implication that every figure() must be paired with close(). Closing a figure window that was created using show() also deregisters the figure from pyplot and hence close() is not needed in that usage scenario.

Something like

> pyplot maintains a reference to figures created with `figure()`. When work on the figure is completed, it should be closed, i.e. deregistered from pyplot, to free its memory (see also :rc:`figure.max_open_warning`).
> Closing a figure window created by `show()` automatically deregisters the figure. For all other use cases, most prominently `savefig()` without `show()`, the figure must be deregistered explicitly using `close()`.

Reply:
sure, done